examples/js/test-js.js

```diff
@@ -22,13 +22,48 @@ import {
 
 const formidableLogo = require('./formidable.png');
 const cppCodeBlock = `#include <iostream>
+#include <cstdlib>
+#include <sstream>
+#include <pthread.h>
 
-int main()
+struct thread_data_t 
+{
+   int  thread_id;
+   std::string message;
+};
+
+void *print_thread_message(void *thread_arg) 
 {
-  auto curried_add = [](int x) -> function<int(int)> { return [=](int y) { return x + y; }; };
+   struct thread_data_t *thread_data;
+   thread_data = (struct thread_data_t *) thread_arg;
+
+   cout << "Thread ID: " << thread_data->thread_id;
+   cout << "Message: " << thread_data->message << endl;
 
-  auto answer = curried_add(7)(8);
-  std::cout << answer << std::endl;
+   pthread_exit(NULL);
+}
+
+int main()
+{
+  pthread_t threads[NUM_THREADS];
+  struct thread_data_t thread_data[NUM_THREADS];
+  
+  for (int i = 0; i < NUM_THREADS; i++) 
+  {
+    auto curried_add = [](int x) -> function<int(int)> { return [=](int y) { return x + y; }; };
+    auto answer = curried_add(i)(5);
+    
+    std::stringstream message;
+    message << "The math result is " << answer << "!";
+    thread_data.thread_id = i;
+    thread_data.message = message.str();
+    int err = pthread_create(&threads, NULL, print_thread_message, (void *)&thread_data[i]);
+    
+    if (err) 
+    {
+      exit(-1)
+    }
+  }
```

Comment:
This is all demo C++ to make the code pane bigger. No need to review this.

Reply:
Are you sure this is thread-safe? :trollface:

Reply:
💅